test/linalg/cholesky.jl

```diff
@@ -66,10 +66,24 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
         capds = cholfact(apds)
         @test inv(capds)*apds ≈ eye(n)
         @test abs((det(capds) - det(apd))/det(capds)) <= ε*κ*n
+        if eltya <: BlasReal
+            capds = cholfact!(copy(apds))
```

Comment:
I believe that @Sacha0's analysis is correct. However, by far most of the time in the linear algebra tests is spent compiling the methods so it's essentially free to run redundant repetitions and when restructuring the dispatching logic of these functions, it can occasionally be useful to have tests that exercise the same path from different entry points os I think it's okay to keep these tests.